network/common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java

```diff
@@ -76,7 +76,13 @@ public void exceptionCaught(Throwable cause) {
 
   @Override
   public void channelUnregistered() {
-    streamManager.connectionTerminated(channel);
+    if (streamManager != null) {
+      try {
+        streamManager.connectionTerminated(channel);
+      } catch (RuntimeException e) {
```

Comment:
Because that's the only exception that can be thrown by the method (this is java, so you have checked exceptions).